server/models/services/AtVersionService.js

```diff
@@ -0,0 +1,460 @@
+ */
+const getRefreshableTestPlanReportsForVersion = async ({
+  currentAtVersionId,
+  transaction
+}) => {
+  const currentVersion = await ModelService.getById(AtVersion, {
+    id: currentAtVersionId,
+    attributes: AT_VERSION_ATTRIBUTES,
+    include: [atAssociation(AT_ATTRIBUTES)],
+    transaction
+  });
+
+  if (!currentVersion) {
+    throw new Error(`AT Version with ID ${currentAtVersionId} not found`);
+  }
+
+  const atName = currentVersion.at?.name;
+  if (!atName) {
+    throw new Error(
+      `AT record is not associated with AT Version ID ${currentAtVersionId}`
+    );
+  }
+
+  // Get all automation-supported AT versions for the same AT, sorted
+  const automationVersions = await AtVersion.findAll({
+    where: {
+      atId: currentVersion.atId,
+      name: { [Op.in]: AT_VERSIONS_SUPPORTED_BY_COLLECTION_JOBS[atName] }
+    },
+    order: [['releasedAt', 'DESC']],
+    transaction
+  });
+
+  if (!automationVersions.length) {
+    return { currentVersion, previousVersion: null, refreshableReports: [] };
+  }
+
+  // Check if the current version is the most recent automation-supported version;
+  const latestAutomationVersion = automationVersions[0];
+  if (currentVersion.id !== latestAutomationVersion.id) {
+    return { currentVersion, previousVersion: null, refreshableReports: [] };
+  }
+
+  // The most recent automation-supported version released before the current one.
+  const previousVersion = automationVersions[1];
+  if (!previousVersion) {
+    return { currentVersion, previousVersion: null, refreshableReports: [] };
+  }
+
+  // Find TestPlanReports that are eligible for refresh based on their TestPlanRun.
+  // Use a raw condition that checks that at least one element in testResults has atVersionId equal to previousVersion.id.
+  const testPlanRunResults = await TestPlanRun.findAll({
+    attributes: ['testPlanReportId'],
+    where: {
+      initiatedByAutomation: true,
+      [Op.and]: Sequelize.literal(`EXISTS (
+        SELECT 1
+        FROM jsonb_array_elements("TestPlanRun"."testResults") AS js
+        WHERE (js->>'atVersionId')::int = ${previousVersion.id}
+      )`)
+    },
+    transaction,
+    raw: true
+  });
+
+  const reportIds = [
+    ...new Set(testPlanRunResults.map(run => run.testPlanReportId))
+  ];
+
+  let refreshableReports = [];
+  if (reportIds.length) {
+    refreshableReports = await TestPlanReport.findAll({
+      where: { id: { [Op.in]: reportIds } },
+      transaction
+    });
+  }
+
+  return { currentVersion, previousVersion, refreshableReports };
+};
+
+module.exports = {
+  getAtVersions,
+  getAtVersionWithRequirements,
+  getAtVersionById,
+  getAtVersionByQuery,
+  createAtVersion,
+  updateAtVersionById,
+  updateAtVersionByQuery,
+  removeAtVersionByQuery,
+  removeAtVersionById,
+  findOrCreateAtVersion,
+  getUniqueAtVersionsForReport,
+  getRefreshableTestPlanReports: getRefreshableTestPlanReportsForVersion
```

Comment:
nit: Is there a reason for not exporting with the function name. Or should the function name be renamed to `getRefreshableTestPlanReports`?

Just wondering if it's because of the sole import location and that context why the rename happened here. If so, I personally would rather the rename happen where the function is being imported because this just causes a tiny bit of initial confusion when looking at these exports.

Reply:
Ah good callout! No this is just due to me renaming the function using my IDE's "auto-rename" feature and the feature making a choice to keep the name the same in the export. Can fix